ios/brave-ios/Sources/Brave/Frontend/Browser/HomePanel/NTPDataSource.swift

```diff
@@ -104,114 +110,116 @@ public class NTPDataSource {
-      switch strategy {
-      case .sponsoredRotation:
-        return Int.random(in: 0..<backgroundSet.count)
-
-      case .randomOrderAvoidDuplicates:
-        let availableRange = 0..<backgroundSet.count
-        // This takes all indeces and filters out ones that were shown recently
-        let availableBackgroundIndeces = availableRange.filter {
-          !self.lastBackgroundChoices.contains($0)
-        }
-        // Due to how many display modes currently exist, the background avoidance counter may get utilized on a smaller subset.
-        // This can be repro by swapping between normal backgrounds and a super referrer, where all available indeces get squeezed out, resulting in an empty set.
-        // To avoid issues, first fallback results in full set.
-
-        // Chooses a new random index to use from the available indeces
-        let chosenIndex =
-          availableBackgroundIndeces.randomElement() ?? availableRange.randomElement() ?? -1
-        assert(chosenIndex >= 0, "NTP index was nil, this is terrible.")
-        assert(chosenIndex < backgroundSet.count, "NTP index is too large, BAD!")
-
-        // This index is now added to 'past' tracking list to prevent duplicates
-        self.lastBackgroundChoices.append(chosenIndex)
-        // Trimming to fixed length to release older backgrounds
-
-        self.lastBackgroundChoices = self.lastBackgroundChoices
-          .suffix(min(backgroundSet.count - 1, NTPDataSource.numberOfDuplicateAvoidance))
-        return chosenIndex
+      let availableRange = 0..<backgroundSet.count
+      // This takes all indeces and filters out ones that were shown recently
+      let availableBackgroundIndeces = availableRange.filter {
+        !self.lastBackgroundChoices.contains($0)
       }
+      // Due to how many display modes currently exist, the background avoidance counter may get utilized on a smaller subset.
+      // This can be repro by swapping between normal backgrounds and a super referrer, where all available indeces get squeezed out, resulting in an empty set.
+      // To avoid issues, first fallback results in full set.
+
+      // Chooses a new random index to use from the available indeces
+      let chosenIndex =
+        availableBackgroundIndeces.randomElement() ?? availableRange.randomElement() ?? -1
+      assert(chosenIndex >= 0, "NTP index was nil, this is terrible.")
+      assert(chosenIndex < backgroundSet.count, "NTP index is too large, BAD!")
+
+      // This index is now added to 'past' tracking list to prevent duplicates
+      self.lastBackgroundChoices.append(chosenIndex)
+      // Trimming to fixed length to release older backgrounds
+
+      self.lastBackgroundChoices = self.lastBackgroundChoices
+        .suffix(min(backgroundSet.count - 1, NTPDataSource.numberOfDuplicateAvoidance))
+      return chosenIndex
     }()
 
+    guard let bgWithIndex = backgroundSet[safe: backgroundIndex] else { return nil }
+    return bgWithIndex
```

Comment:
No need to guard here unless you're going to use `bgWithIndex`

```suggestion
    return backgroundSet[safe: backgroundIndex]
```

Reply:
Fixed